apps/forklift/lib/forklift/data_writer.ex

```diff
@@ -113,15 +114,16 @@ defmodule Forklift.DataWriter do
   defp do_write(data, dataset) do
     started_data = Enum.map(data, &add_start_time/1)
 
-    retry with: exponential_backoff(100) |> cap(@max_wait_time) do
+    retry with: exponential_backoff(@retry_count) |> expiry(@max_wait) do
```

Comment:
```suggestion
    retry with: exponential_backoff(100) |> cap(@max_wait) |> Stream.take(@retry_count) do
```

Reply:
I merged in this suggestion change. Unit tests, integration tests, and e2e all pass.